cmd/maintained-apps/validate/main.go

```diff
@@ -0,0 +1,326 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"slices"
+	"strings"
+	"time"
+
+	maintained_apps "github.com/fleetdm/fleet/v4/ee/maintained-apps"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	mdm_maintained_apps "github.com/fleetdm/fleet/v4/server/mdm/maintainedapps"
+)
+
+var (
+	tmpDir                      string
+	env                         []string
+	installationSearchDirectory string
+	operatingSystem             string
+)
+
+func run() int {
+	operatingSystem = strings.ToLower(os.Getenv("GOOS"))
+	if operatingSystem == "" {
+		operatingSystem = runtime.GOOS
+		fmt.Printf("GOOS environment variable is not set. Using system detected: '%s'\n", operatingSystem)
```

Comment:
Here and elsewhere, I think it's worth replacing the printfs with a dedicated logger. Like you mentioned on the call, we could even have a nice toggle for verbosity.